### DISCUSSION TITLE
v0.40.2

### DISCUSSION THREAD

**Discussion by rfratto** [grafana-cold-storage/agent#6624] [Announcements]
This is release `v0.40.2` of Grafana Agent.

### Upgrading

Read the relevant upgrade guides for specific instructions on upgrading from older versions:

* [Static mode upgrade guide](https://grafana.com/docs/agent/v0.40/static/upgrade-guide/)
* [Static mode Kubernetes operator upgrade guide](https://grafana.com/docs/agent/v0.40/operator/upgrade-guide/)
* [Flow mode upgrade guide](https://grafana.com/docs/agent/v0.40/flow/upgrade-guide/)

#### Bugfixes

* Set permissions on the Grafana Agent [Flow] folder when installing via the windows installer rather than relying on the parent folder permissions. (@erikbaranowski)
* Set restricted viewing permissions on the agent-config.yaml (static mode) or config.river (flow mode) when installing via the Windows installer if the configuration file does not already exist. (@erikbaranowski)
* Fix an issue where the import config node would not run after a config reload. (@wildum) 
* Fix an issue where Loki could reject a batch of logs when structured metadata feature is used. (@thampiotr)
* Fix a duplicate metrics registration panic when recreating static mode metric instance's write handler. (@rfratto, @hainenber)

#### Other changes

 * Change the Docker base image for Linux containers to public.ecr.aws/ubuntu/ubuntu:mantic. (@hainenber)

### Installation

Refer to our installation guides for how to install the variants of Grafana Agent:

* [Install static mode](https://grafana.com/docs/agent/v0.40/static/set-up/install/)
* [Install the static mode Kubernetes operator](https://grafana.com/docs/agent/v0.40/operator/helm-getting-started/)
* [Install flow mode](https://grafana.com/docs/agent/v0.40/flow/setup/install/)

<hr /><em>This discussion was created from the release <a href='https://github.com/grafana/agent/releases/tag/v0.40.2'>v0.40.2</a>.</em>

**Comment by madridu** [comment]
This and previous version are been marked as infected by ESET Antivirus since the begining of the week. Checked with others (AVAST, McAfee, etc) and they're also marking agent as infected by trojan.wingo.

**Reply by rfratto** [reply]
Can you let me know which binary is being flagged, and what the SHA of that binary that is being flagged is?

**Reply by rfratto** [reply]
Looking around a bit, it sounds like binaries built with Go 1.22 are flagged as [false positives](https://forum.eset.com/topic/39844-false-positive-go-1220/); it sounds like you'll need to update your definitions, or if they're up to date, wait for other AVs to update their definitions.

**Reply by madridu** [reply]
The binary is grafana-agent-flow-windows-amd64.exe with hash B8372463FA49A0379EBEC6F85E64B5B6E98EFFF7.

We thought it was a false positive. We have our definitions up to date, it was working until last tuesday I guess it was an update of definitions which caused it. We'll wait to see if a new update solves it
